include/deal.II/lac/linear_operator.h

```diff
@@ -1111,6 +1111,130 @@ linear_operator(const OperatorExemplar &operator_exemplar, const Matrix &matrix)
   return return_op;
 }
 
+template <typename Range = Vector<double>,
+          typename Domain = Range,
+          typename Matrix>
+LinearOperator<Range, Domain>
+constraints_linear_operator(const ConstraintMatrix &, const Matrix &);
+
+/**
+ * @relates LinearOperator
+ *
+ * constraints_linear_operator takes a constraint matrix @p cm and an exemplar
+ * matrix @p m (this matrix could be the system matrix).
+ * The result of this function is the LinearOperator associated to the constraint
+ * matrix.
+ * In detail, a constrained problem can be expressed in this form:
+ * Ct * S * C * x = Ct * ( rhs + S * k)
+ * Where:
+ *  - C has the role of constraint matrix
```

Comment:
I think you should still explain how this C relates to the constraints and inhomogeneities stored by the ConstraintMatrix class.

Specifically, the ConstraintMatrix class thinks of itself as a matrix of size $m x n$ where m=number of constraints, and n=number of dofs. On the other hand, you tend to think of C as an $n x n$ matrix.